src/Chain.cpp

```diff
@@ -8,93 +7,67 @@
 using namespace Xoroshiro;
 
 Chain::Chain(QSize size)
-    :m_openGLWorker(new ChainOpenGLWorker(this))
-    , m_size(size) {
-    connect(m_openGLWorker, &ChainOpenGLWorker::initialized, this, &Chain::onInitialized);
-    connect(this, &QObject::destroyed, m_openGLWorker, &QObject::deleteLater);
-
-    if (!QMetaObject::invokeMethod(m_openGLWorker, "initialize", Q_ARG(QSize, m_size)))
-        qFatal("Unable to initialize openGLWorker");
+    : m_noiseTexture(nullptr)
+    , m_blankTexture(nullptr)
+    , m_vao(nullptr)
+    , m_size(size)
+    {
+        m_noiseTexture = new QOpenGLTexture(QOpenGLTexture::Target2D);
+        m_blankTexture = new QOpenGLTexture(QOpenGLTexture::Target2D);
+        m_vao = new QOpenGLVertexArrayObject();
 }
 
-Chain::~Chain() = default;
-
-void Chain::onInitialized(int n, int b) {
-    m_noiseTextureId = n;
-    m_blankTextureId = b;
-    m_initialized = true;
+Chain::~Chain() {
+    // TODO XXX destroy the noise texture and the blank texture
+    delete m_vao;
 }
 
 QSize Chain::size() {
     return m_size;
 }
 
-qreal Chain::beatTime() const {
-    return m_beatTime;
-};
-qreal Chain::realTime() const {
-    return m_realTime;
-};
-void Chain::setBeatTime(qreal _time) {
-    m_beatTime = _time;
-}
-void Chain::setRealTime(qreal _time) {
-    m_realTime = _time;
-}
-GLuint Chain::noiseTexture() const {
-    return m_noiseTextureId;//.textureId();
-}
+GLuint Chain::noiseTexture() {
```

Comment:
Yeah, this is intentional. It is super useful as a PRNG (we should probably use this instead of the `random` function.) A lot of shadertoy effects require a static buffer of noise.